app/services/invoices/payments/cashfree_service.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+module Invoices
+  module Payments
+    class CashfreeService < BaseService
+      include Customers::PaymentProviderFinder
+
+      PENDING_STATUSES = %w[PARTIALLY_PAID].freeze
+      SUCCESS_STATUSES = %w[PAID].freeze
+      FAILED_STATUSES = %w[EXPIRED CANCELLED].freeze
+
+      def initialize(invoice = nil)
+        @invoice = invoice
+
+        super(nil)
+      end
+
+      def create
+        Rails.logger.debug "invoice.total_amount_cents 0: #{invoice.total_amount_cents}"
+
+        result.invoice = invoice
+        return result unless should_process_payment?
+
+        Rails.logger.debug "invoice.total_amount_cents 1: #{invoice.total_amount_cents.positive?}"
+
+        unless invoice.total_amount_cents.positive?
+          update_invoice_payment_status(payment_status: :succeeded)
+          return result
+        end
+
+        Rails.logger.debug "invoice.total_amount_cents 2: #{invoice.total_amount_cents}"
+
+        increment_payment_attempts
+
+        payment = Payment.new(
+          payable: invoice,
+          payment_provider_id: cashfree_payment_provider.id,
+          payment_provider_customer_id: customer.cashfree_customer.id,
+          amount_cents: invoice.total_amount_cents,
+          amount_currency: invoice.currency.upcase,
+          provider_payment_id: invoice.id,
+          status: :pending
+        )
+        payment.save!
+
+        result.payment = payment
+        result
+      end
+
+      def update_payment_status(provider_payment_id:, status:, metadata: {})
+        payment = if metadata[:payment_type] == 'one-time'
```

Comment:
With stripe integration, this metadata value is passed when creating the payment on stripe. Does it apply the same way here?

Reply:
Yes, you are correct. I am not using the `metadata` because I am creating the `Payment` object using the `create` method. Therefore, I can safely remove it along with the `create_payment` method.

Reply:
This method is used to update the status of the payment inside Lago's database when receiving a webhook from the third party service. It means that the format of the metadata even its presence will depend of each service.